alibuild_helpers/build.py

```diff
@@ -878,6 +886,7 @@ def downloadTask(p):
       if spec["devel_hash"]+spec["deps_hash"] == spec["old_devel_hash"]:
         info("Development package %s does not need rebuild", spec["package"])
         buildOrder.pop(0)
+        packageIterations = 0
```

Comment:
What is this for?

Reply:
I think this was forgotten at some point -- this tracks the number of times the current package was rebuilt, but we've just decided we're done with the current package, so `packageIterations` should be reset.

As a side note, this was/is really unlikely to actually cause a bug because the limit of iterations is set at 20, so one or two more for the next package still won't trip the limit. (Which is why it wasn't noticed before, I presume.)

Reply:
Reverted; I'll include it in the cleanup PR.